client/packages/cli/index.js

```diff
@@ -1070,10 +1071,27 @@ async function pushPerms(appId) {
     return;
   }
 
-  const ok = await promptOk(
-    "Pushing permissions rules. This will immediately replace your production rules. OK to proceed?",
-  );
-  if (!ok) return;
+  console.log("Planning...");
+
+  const prodPerms = await fetchJson({
+    path: `/dash/apps/${appId}/perms/pull`,
+    debugName: "Perms pull",
+    errorMessage: "Failed to pull perms.",
+  });
+
+  if (!prodPerms.ok) return;
+
+  const diffedStr = jsonDiff.diffString(prodPerms.data.perms, perms)
+  if (!diffedStr.length) {
+    console.log("There are no changes to apply.");
```

Comment:
Will it be obvious to the user that we're talking about perms changes here?

Reply:
Ah good point! Will update the text 🫡